src/main/java/com/thoughtworks/agenciacompromisso/controllers/UserController.java

```diff
@@ -0,0 +1,32 @@
+package com.thoughtworks.agenciacompromisso.controllers;
+
+import com.thoughtworks.agenciacompromisso.services.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+
+@RestController
+@RequestMapping("/authorize")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @RequestMapping("/{email}")
+    @GetMapping
+    public ResponseEntity authorize(@PathVariable("email") String email) {
+        Boolean isAuthorized = userService.isAuthorized(email);
+        if (isAuthorized) {
+            HttpHeaders headers = new HttpHeaders();
+            headers.add("Location", "http://localhost:3000/menu");
```

Comment:
Aqui é importante não passar o localhost pois depende do ambiente.
Apesar que nesse caso você está passando uma url do client. Acho que a ideia é a API não saber pra onde deve redirecionar quando for autorizado. Nesse caso acho legal só retornar se foi ou não autorizado e o frontend fazer o redirect.